### PR TITLE
Added Hackathon CRUD API Views

### PR DESCRIPTION
**Pull request by shubhanshu02**
The following changes are proposed with this PR.

1. Added Generic CreateAPIView for hackathon object creation
2. Added Generic RetrieveUpdateDestroyAPIView for accessing and modifying hackathons with their id
3. Added Hackathon Serializer
4. Ran `manage.py makemigrations` to remove date and max_score fields from the hackathon table.

### CODE REVIEW

File: core/views.py [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+        1. GET can be accessed even by anonymous users
+        2. PUT, PATCH and DELETE can only be accessed by the Super User
+        '''
+        def has_permission(self, request, view):
+            if (request.method in ['PUT','PATCH','DELETE']):
+                return bool(request.user and request.user.is_superuser)
+            elif (request.method == 'GET'):
+                return True
+            else:
+                return False
```
Comment (nishantwrp) [inline core/views.py]: You can use `SAFE_METHODS` in django-rest-framework package here. And define permissions in a separate `permissions.py` file.

**Review by nishantwrp — CHANGES_REQUESTED** [review]
lgtm just one comment.

File: core/urls.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from django.urls import path
+from .views import (HackathonsRUDView,HackathonCreateView)
+
+urlpatterns = [   
+    path('hackathons/', HackathonCreateView.as_view(), name='Hackathon Create View'),
+    path('hackathons/<int:id>', HackathonsRUDView.as_view(), name='Hackathon Read, Edit and Delete View'),
```
Comment (nishantwrp) [inline core/urls.py]:
```suggestion
    path('hackathons/<int:pk>', HackathonsRUDView.as_view(), name='Hackathon Read, Edit and Delete View'),
```

File: core/views.py [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+    
+    permission_classes = [HackathonPermissions]
+    serializer_class = HackathonSerializer
+    lookup_url_kwarg = 'id'
```
Comment (nishantwrp) [inline core/views.py]:
```suggestion
    lookup_field = 'pk'
```

File: core/views.py (hunk 1 above)
Comment (shubhanshu02) [reply core/views.py]: Done

**Review by nishantwrp — APPROVED** [review]
lgtm

**Pull request merged**